src/Language/Haskell/TH/Syntax/Compat.hs

```diff
@@ -768,6 +769,20 @@ bindSplice_ = bindCode_
 bindSplice_ q c = liftSplice ( q >> examineSplice c)
 # endif
 
+-- | Lift a @'Syntax.TExp' a@ into a 'Splice'. This is useful when splicing
+-- in the result of a computation into a typed QuasiQuoter.
+--
+-- @
+-- foo :: 'SpliceQ' a
+-- foo = 'liftSplice' $ do
+--      ints <- [|| [1,2,3] ||] :: SpliceQ [Int]
```

Comment:
As far as I can tell, this code doesn't compile on GHC 9.0+, since `SpliceQ = Code Q` and `Code` isn't a `Monad`. Perhaps you meant this?

```suggestion
--      ints <- examineSplice ([|| [1,2,3] ||] :: SpliceQ [Int])
```

Having written that out, however, it strikes me that this example might be a bit too simplistic, since it's arguably easier to just write `let ints = [|| [1,2,3] ||] in examineSplice [|| sum $$(ints) ||]`. To properly motivate the need for `expToSplice`, we might need a slightly more involved example, such as the use of `listTE` in the code you added to the `README`. I'll leave that choice to your discretion.

Reply:
yeah! I'll bring `listTE` in in a minimal way.